lib/manager/terraform/extract.ts

```diff
@@ -23,52 +24,80 @@ export function getTerraformDependencyType(
     case 'provider': {
       return TerraformDependencyTypes.provider;
     }
+    case 'required_providers': {
+      return TerraformDependencyTypes.required_providers;
+    }
     default: {
       return TerraformDependencyTypes.unknown;
     }
   }
 }
 
+const dependencyBlockExtractionRegex = /^\s*(?<type>module|provider|required_providers)\s+("(?<lookupName>[^"]+)"\s+)?{\s*$/;
+const keyValueExtractionRegex = /^\s*(?<key>[^\s]+)\s+=\s+"(?<value>[^"]+)"\s*$/; // extracts `exampleKey = exampleValue`
+
 export function extractPackageFile(content: string): PackageFile | null {
   logger.trace({ content }, 'terraform.extractPackageFile()');
-  if (!content.includes('module "') && !content.includes('provider "')) {
+  if (
+    !content.includes('module "') &&
+    !content.includes('provider "') &&
+    !content.includes('required_providers ')
+  ) {
     return null;
   }
   const deps: PackageDependency[] = [];
   try {
     const lines = content.split('\n');
     for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
       let line = lines[lineNumber];
-      const terraformDependency = /^(module|provider)\s+"([^"]+)"\s+{\s*$/.exec(
-        line
-      );
+      const terraformDependency = dependencyBlockExtractionRegex.exec(line);
       if (terraformDependency) {
-        logger.trace(`Matched ${terraformDependency[1]} on line ${lineNumber}`);
-        const tfDepType: TerraformDependencyTypes = getTerraformDependencyType(
-          terraformDependency[1]
+        logger.trace(
+          `Matched ${terraformDependency.groups.type} on line ${lineNumber}`
+        );
+        const tfDepType = getTerraformDependencyType(
+          terraformDependency.groups.type
         );
-        const dep: PackageDependency = {
-          managerData: {
-            moduleName: terraformDependency[2],
-            terraformDependencyType: tfDepType,
-          },
-        };
+
         if (tfDepType === TerraformDependencyTypes.unknown) {
           /* istanbul ignore next */ logger.trace(
-            `Could not identify TerraformDependencyType ${terraformDependency[1]} on line ${lineNumber}.`
+            `Could not identify TerraformDependencyType ${terraformDependency.groups.type} on line ${lineNumber}.`
           );
+        } else if (tfDepType === TerraformDependencyTypes.required_providers) {
+          do {
+            const dep: PackageDependency = {
+              managerData: {
+                terraformDependencyType: tfDepType,
+              },
+            };
+
+            lineNumber += 1;
+            line = lines[lineNumber];
+            const kvMatch = keyValueExtractionRegex.exec(line);
+            if (kvMatch) {
+              dep.currentValue = kvMatch.groups.value;
+              dep.managerData.moduleName = kvMatch.groups.key;
+              dep.managerData.versionLine = lineNumber;
```

Comment:
@rarkins do we need this? I think it is obsolete.

Reply:
Yes, no longer needed

Reply:
So there is no need for any kind of propagation of the line number?